### ISSUE TITLE
Replace pulse drawer with v2

### ISSUE THREAD

**Issue by nkanazawa1989** [Qiskit/qiskit-terra#5392]
### What is the expected enhancement?

Pulse drawer v2 is merged with #5101, and now we can replace the default pulse drawer with new program. Before that, we need to do sufficient usability test. This drawer has backward compatibility and thus it is easy to deprecate current drawer.

New drawer can be called with
```python
from qiskit import circuit, transpile, schedule
from qiskit.visualization.pulse_v2 import draw as pulse_drawer
from qiskit.test.mock import FakeAlmaden

qc = circuit.QuantumCircuit(2, 2)
qc.h(0)
qc.cx(0, 1)

pulse_drawer(schedule(transpile(qc, FakeAlmaden()), FakeAlmaden()))
```

**Comment by lcapelluto** [comment]
The new drawer is so much better:) The old one is almost unusable, with scaling issues, so I added this to 0.17 milestone. Really looking forward to this!

It doesn't seem to have full backwards compatibility as is. I get errors:
`TypeError: draw() got an unexpected keyword argument 'dt'`
`TypeError: draw() got an unexpected keyword argument 'filename'`
etc

**Comment by nkanazawa1989** [comment]
Thanks for testing new drawer. The changes are below:
- As for `dt`: we have `time_unit` and `backend` as new arguments. `dt` value is automatically extracted from given pulse backend object. We can set `dt` or `ns` to `time_unit` argment.
- As for `filename`: We delegated file saving to the plotter software to be used. The pulse drawer returns matplotlib `Figure` object and we can call `Figure.savefig` method with lots of figure saving options. Removing this coupling is necessary to support multiple APIs in future.

I've found a few bugs now and fix is here https://github.com/Qiskit/qiskit-terra/pull/5425
Once this is merged, we can perhaps replace current drawer with v2.
